lib/daemon_runner/semaphore.rb

```diff
@@ -19,17 +19,19 @@ def lock(name, limit = 3, **options)
         semaphore = Semaphore.new(options)
         semaphore.lock
         if block_given?
-          lock_thr = semaphore.renew
-          yield
+          begin
+            lock_thr = semaphore.renew
+            yield
+          ensure
+            lock_thr.kill
```

Comment:
I don't think [`Thread.new` can return `nil`](https://github.com/rapid7/daemon_runner/blob/0.4.1/lib/daemon_runner/semaphore.rb#L85), so we should be okay.